CRM/MembershipExtras/Service/MembershipEndDateCalculator.php

```diff
@@ -9,15 +12,68 @@ class CRM_MembershipExtras_Service_MembershipEndDateCalculator {
    * @param int $membershipId
    *
    * @return string
+   * @throws \CiviCRM_API3_Exception
+   * @throws \CiviCRM_API3_Exception|\Exception
    */
   public static function calculate($membershipId) {
-    $membershipDetails = civicrm_api3('Membership', 'get', [
+    $newEndDate = 'null';
```

Comment:
@MiyaNoctem  'null' not NULL or ''?

Reply:
Yeah, has to be `'null'` so CiviCRM sets it as NULL in the DB. `NULL` will not add it to the query, and `''` will set it to an empty string.